DEPLOY.rst

```diff
@@ -0,0 +1,198 @@
+Deploying to AWS with autoscaling
+=================================
+
+This deploy system makes heavy use of
+`Fabulaws <http://fabulaws.readthedocs.org/en/latest/index.html>`_.
+Please at least skim those docs first.
+
+Note that there are a few changes in this project from the
+default architecture described in the Fabulaws docs:
+
+* We use django-pipeline instead of django-compressor.
+* We serve the static files from the Django processes in
+  our web servers instead of using S3.
+
+(These might change, especially the not using S3 part, but they're
+correct at the time this is written.)
+
+Terms
+-----
+
+These are some terms that'll be used in the deploy files.
+
+*Deployment:* The project. In this case, ``opendebates``.
+
+*Environment:* A configuration and set of one or more systems that
+make the project available on the network with a unique domain and
+set of data.  For example, we might have testing, staging,
+and production environments.
+
+*Role:* Each system in an environment plays one or more roles, meaning
+that it provides some service to the environment. Example roles include
+db-master, db-slave, cache, web, worker, etc.
+
+From scratch
+------------
+
+AWS console
+~~~~~~~~~~~
+
+Currently there's some manual setup required in the AWS console
+before we can start using the automation.
+
+Please work *carefully and thoroughly* through the
+`Fabulaws initial setup documentation <http://fabulaws.readthedocs.org/en/latest/initial-setup.html>`_, changing
+``myproject`` to ``opendebates``.  This will include creating security groups,
+creating a load balancer, and creating an autoscaling group.
+
+You can skip most of the *Project Configuration* section because that's
+already been done for this project, except for
+`SSH Keys <http://fabulaws.readthedocs.org/en/latest/initial-setup.html#ssh-keys>`_.
+
+Changes to project files
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In ``fabulaws-config.yml`` in this project, find the section that
+looks like this::
+
+    site_domains_map:
+      production:
+      - dualstack.myproject-production-1-12345.us-east-1.elb.amazonaws.com
+      staging:
+      - dualstack.myproject-staging-1-12345.us-east-1.elb.amazonaws.com
+      testing:
+      - dualstack.myproject-testing-1-12345.us-east-1.elb.amazonaws.com
+
+and change the domain under the environment you're setting up to the
+full hostname of the load balancer you just created.
+
+Also look for this section::
+
+      auto_scaling_groups:
+        opendebates:
+          production: opendebates-production-ag
+          staging: opendebates-staging-ag
+          testing: opendebates-testing-ag
+
+and change the appropriate value to the name of the autoscaling group you
+created.  (Or create it with that name to begin with.)
+
+Automation commands
+~~~~~~~~~~~~~~~~~~~
+
+Now you can run this command to set up everything else::
```

Comment:
Most people will never run this command; we'll set it up once and then keep it live. Might be good to mention that here or lower this section in the docs.